xdsl/ir.py

```diff
@@ -424,9 +424,11 @@ def verify(self, verify_nested_ops: bool = True) -> None:
     def verify_(self) -> None:
         pass
 
+    _OperationType = TypeVar('_OperationType', bound='Operation')
+
     @classmethod
-    def parse(cls: type[Parser._OperationType], result_types: list[Attribute],
-              parser: Parser) -> Parser._OperationType:
+    def parse(cls: type[_OperationType], result_types: list[Attribute],
```

Comment:
This was previously the a TypeVar shared with the Parser right? That's what has changed here?

I am not quite sure about the implications here. Maybe @math-fehr them better.

Reply:
I personally prefer the new version, since this won't get a private variable from another file.